From: Egor Bychin <e.bychin@drweb.com>
Date: Mon, 11 Oct 2021 14:08:34 +0300
Subject: gbacktrace: Fix fcntl command arguments not being checked

Origin: upstream, 2.70.1, commit:4d8c5b6d77ed5da355d6c59465dffa2ad5648894
---
 glib/gbacktrace.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/glib/gbacktrace.c b/glib/gbacktrace.c
index 6181073..eb9ec8d 100644
--- a/glib/gbacktrace.c
+++ b/glib/gbacktrace.c
@@ -340,7 +340,11 @@ stack_trace (const char * const *args)
       /* Save stderr for printing failure below */
       int old_err = dup (2);
       if (old_err != -1)
-        fcntl (old_err, F_SETFD, fcntl (old_err, F_GETFD) | FD_CLOEXEC);
+	{
+	  int getfd = fcntl (old_err, F_GETFD);
+	  if (getfd != -1)
+	    (void) fcntl (old_err, F_SETFD, getfd | FD_CLOEXEC);
+	}
 
       close (0); dup (in_fd[0]);   /* set the stdin to the in pipe */
       close (1); dup (out_fd[1]);  /* set the stdout to the out pipe */
