From: Egor Bychin <e.bychin@drweb.com>
Date: Mon, 11 Oct 2021 13:54:07 +0300
Subject: gdbusauthmechanismsha1: Fix pointer being freed after being NULLed

Origin: upstream, 2.70.1, commit:5796babc5cd2ea315d3b3771310ee4734c89dda6
---
 gio/gdbusauthmechanismsha1.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gio/gdbusauthmechanismsha1.c b/gio/gdbusauthmechanismsha1.c
index 095a666..94fe0bc 100644
--- a/gio/gdbusauthmechanismsha1.c
+++ b/gio/gdbusauthmechanismsha1.c
@@ -881,8 +881,8 @@ keyring_generate_entry (const gchar  *cookie_context,
                                      error))
         {
           *out_id = 0;
-          *out_cookie = 0;
           g_free (*out_cookie);
+          *out_cookie = 0;
           ret = FALSE;
           goto out;
         }
