From: Uwe Korn <gnome@uwekorn.com>
Date: Thu, 11 Sep 2025 16:04:15 +0000
Subject: `gio/tests/socket-listener` should depend on `libdl`

Bug: https://gitlab.gnome.org/GNOME/glib/-/issues/3780
Origin: upstream, 2.86.1, commit:2b0f5304bdf8804855ca3831d9ac265038442c20
---
 gio/tests/meson.build | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/gio/tests/meson.build b/gio/tests/meson.build
index 32ebda3..0044121 100644
--- a/gio/tests/meson.build
+++ b/gio/tests/meson.build
@@ -129,7 +129,9 @@ gio_tests = {
     # FIXME: https://gitlab.gnome.org/GNOME/glib/-/issues/3148
     'can_fail' : host_system in ['darwin', 'gnu'],
   },
-  'socket-listener' : {},
+  'socket-listener' : {
+    'dependencies': libdl_dep,
+  },
   'socket-service' : {},
   'srvtarget' : {},
   'task' : {},
