From: Egor Bychin <e.bychin@drweb.com>
Date: Mon, 11 Oct 2021 14:10:01 +0300
Subject: gkeyfile: Remove unreachable code

Origin: upstream, 2.70.1, commit:da1d64cb4d6d6a7faa12583c16f332964bce95a5
---
 glib/gkeyfile.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/glib/gkeyfile.c b/glib/gkeyfile.c
index 56500cc..17cf856 100644
--- a/glib/gkeyfile.c
+++ b/glib/gkeyfile.c
@@ -2259,9 +2259,6 @@ g_key_file_get_locale_string (GKeyFile     *key_file,
 
       if (translated_value)
 	break;
-
-      g_free (translated_value);
-      translated_value = NULL;
    }
 
   /* Fallback to untranslated key
