From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Fri, 10 May 2024 01:17:41 +0200
Subject: glib/tests/mapping: Unref the mapped file on exit

Origin: upstream, 2.80.3, commit:9c611f6ef94fce218172c2f22b40d34b6bdbed1d
---
 glib/tests/mapping.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/glib/tests/mapping.c b/glib/tests/mapping.c
index 5fcde83..bbd4de8 100644
--- a/glib/tests/mapping.c
+++ b/glib/tests/mapping.c
@@ -125,6 +125,7 @@ child_main (void)
   g_free (childname);
   g_free (global_filename);
   g_free (dir);
+  g_mapped_file_unref (map);
 
   signal_parent (NULL);
 }
