From: Egor Bychin <e.bychin@drweb.com>
Date: Mon, 11 Oct 2021 13:55:17 +0300
Subject: glocalfile: Fix g_stat return value not being checked

Origin: upstream, 2.70.1, commit:7b582a9993aacfff074cb91187ad406c7755f01b
---
 gio/glocalfile.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/gio/glocalfile.c b/gio/glocalfile.c
index 0d4a44f..da38ade 100644
--- a/gio/glocalfile.c
+++ b/gio/glocalfile.c
@@ -2011,7 +2011,16 @@ g_local_file_trash (GFile         *file,
        * trying to rename across a filesystem boundary, which doesn't work. So
        * we use g_stat here instead of g_lstat, to know where the symlink
        * points to. */
-      g_stat (path, &file_stat);
+      if (g_stat (path, &file_stat))
+	{
+	  errsv = errno;
+	  g_free (path);
+
+	  g_set_io_error (error,
+			  _("Error trashing file %s: %s"),
+			  file, errsv);
+	  return FALSE;
+	}
       g_free (path);
     }
 
