From: Egor Bychin <e.bychin@drweb.com>
Date: Mon, 11 Oct 2021 14:00:03 +0300
Subject: gopenuriportal: Fix GVariantBuilder and string leakage on g_open
 failure

Origin: upstream, 2.70.1, commit:b293fc537614cb6f62810580d823209241f1ec72
---
 gio/gopenuriportal.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gio/gopenuriportal.c b/gio/gopenuriportal.c
index be68569..6ef8f03 100644
--- a/gio/gopenuriportal.c
+++ b/gio/gopenuriportal.c
@@ -108,6 +108,8 @@ g_openuri_portal_open_uri (const char  *uri,
       errsv = errno;
       if (fd == -1)
         {
+	  g_free (path);
+	  g_variant_builder_clear (&opt_builder);
           g_set_error (error, G_IO_ERROR, g_io_error_from_errno (errsv),
                        "Failed to open '%s'", path);
           return FALSE;
