From: Seppi Wilhelmi <wilhelmi@google.com>
Date: Tue, 31 Oct 2023 16:53:40 +0100
Subject: Make GQuark register intentional leaks

Origin: upstream, 2.78.2, commit:7713f376785f897cc31b103bf0538b8d8b76ab6a
---
 glib/gquark.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/glib/gquark.c b/glib/gquark.c
index 42cb393..12ec34c 100644
--- a/glib/gquark.c
+++ b/glib/gquark.c
@@ -43,6 +43,7 @@
 #include "gtestutils.h"
 #include "glib_trace.h"
 #include "glib-init.h"
+#include "glib-private.h"
 
 #define QUARK_BLOCK_SIZE         2048
 #define QUARK_STRING_BLOCK_SIZE (4096 - sizeof (gsize))
@@ -301,6 +302,7 @@ quark_new (gchar *string)
        * us to do lockless lookup of the arrays, and there shouldn't be that
        * many quarks in an app
        */
+      g_ignore_leak (g_atomic_pointer_get (&quarks));
       g_atomic_pointer_set (&quarks, quarks_new);
     }
 
