From: Philip Withnall <pwithnall@gnome.org>
Date: Thu, 18 Sep 2025 16:16:14 +0100
Subject: tests: Rename a couple of tests for consistency

Signed-off-by: Philip Withnall <pwithnall@gnome.org>
Origin: upstream, 2.86.1, commit:0a0da8b21a407f39e9d2e32c30e38bcd0e177947
---
 glib/tests/utils.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/glib/tests/utils.c b/glib/tests/utils.c
index 0423bfc..5eb1492 100644
--- a/glib/tests/utils.c
+++ b/glib/tests/utils.c
@@ -820,7 +820,7 @@ test_hostname (void)
 }
 
 static void
-test_special_dir (void)
+test_user_special_dirs (void)
 {
   const gchar *dir, *dir2;
 
@@ -832,7 +832,7 @@ test_special_dir (void)
 }
 
 static void
-test_desktop_special_dir (void)
+test_user_special_dirs_desktop (void)
 {
   const gchar *dir, *dir2;
 
@@ -1365,8 +1365,8 @@ main (int   argc,
   g_test_add_func ("/utils/username", test_username);
   g_test_add_func ("/utils/realname", test_realname);
   g_test_add_func ("/utils/hostname", test_hostname);
-  g_test_add_func ("/utils/specialdir", test_special_dir);
-  g_test_add_func ("/utils/specialdir/desktop", test_desktop_special_dir);
+  g_test_add_func ("/utils/user-special-dirs", test_user_special_dirs);
+  g_test_add_func ("/utils/user-special-dirs/desktop", test_user_special_dirs_desktop);
   g_test_add_func ("/utils/user-special-dirs/load-unlocked", test_user_special_dirs_load_unlocked);
   g_test_add_func ("/utils/os-info", test_os_info);
   g_test_add_func ("/utils/clear-pointer", test_clear_pointer);
