From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Sat, 11 May 2024 01:23:34 +0200
Subject: vsaprintf: Use proper size for mp_limb_t to please msys2-mingw32 CI

Origin: upstream, 2.80.3, commit:2f35c32c34020d2dad311758d02500e35ff2619c
---
 glib/gnulib/vasnprintf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/glib/gnulib/vasnprintf.c b/glib/gnulib/vasnprintf.c
index a5b68db..7e151b4 100644
--- a/glib/gnulib/vasnprintf.c
+++ b/glib/gnulib/vasnprintf.c
@@ -379,7 +379,7 @@ multiply (mpn_t src1, mpn_t src2, mpn_t *dest)
     {
       /* src1 or src2 is zero.  */
       dest->nlimbs = 0;
-      dest->limbs = (mp_limb_t *) malloc (1);
+      dest->limbs = (mp_limb_t *) malloc (sizeof (mp_limb_t));
     }
   else
     {
